internal/p2p/peermanager.go

```diff
@@ -815,21 +929,83 @@ func (m *PeerManager) Advertise(peerID types.NodeID, limit uint16) []NodeAddress
 		addresses = append(addresses, m.options.SelfAddress)
 	}
 
-	for _, peer := range m.store.Ranked() {
+	var numAddresses int
+	var attempts int
+
+	ranked := m.store.Ranked()
```

Comment:
If `len(ranked) < limit` can we just directly loop through and remove private peers and return early with those addresses

Reply:
yup, I think my latest attempt does this.